File: Project/SpringBoot/src/main/java/com/catcare/project/Service/VeterinarioService.java
package com.catcare.project.Service;

import java.util.Collection;

import com.catcare.project.Entity.Administrador;
import com.catcare.project.Entity.Cliente;
import com.catcare.project.Entity.Veterinario;

public interface VeterinarioService {

    // Obtener todos los veterinarios
    public Collection<Veterinario> SearchAll();

    // Buscar un veterinario por su ID
    public Veterinario SearchById(Long id);

    // Buscar un veterinario por su número de cédula
    public Veterinario findByCedula(String cedula);

    // Agregar un nuevo veterinario
    public void add(Veterinario veterinario);

    // Eliminar un veterinario por su ID
    public void deleteById(Long id);

    // Actualizar los datos de un veterinario
    public void update(Veterinario veterinario);

    // Devuelve la cantidad de veterinarios activos
    public Long countVeterinariosActivos();

    // Devuelve la cantidad de veterinarios activos
    public Long countVeterinariosInactivos();

}

File: Project/Angular/src/app/service/veterinario/veterinario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {Cliente} from "../../cliente/cliente";
import {Veterinario} from "../../veterinario/veterinario";
import {User} from "../../model/user";

@Injectable({
  providedIn: 'root'
})
export class VeterinarioService {

  private ROOT_URL = 'http://localhost:8090/catcare/veterinarios';

  constructor(private http: HttpClient) {}

  // Obtener todos los veterinarios
  getAllVeterinarios(): Observable<any[]> {
    return this.http.get<any[]>(`${this.ROOT_URL}/all`);
  }

  // Obtener un veterinario por su ID
  getVeterinarioById(veterinarioId: number): Observable<any> {
    return this.http.get<any>(`${this.ROOT_URL}/find?id=${veterinarioId}`);
  }

  // Agregar un nuevo veterinario
  agregarVeterinario(veterinario: any): Observable<any> {
    return this.http.post<any>(`${this.ROOT_URL}/agregar`, veterinario);
  }

  // Eliminar un veterinario por su ID
  eliminarVeterinario(veterinarioId: number): Observable<any> {
    return this.http.delete<any>(`${this.ROOT_URL}/delete/${veterinarioId}`);
  }

  // Actualizar los datos de un veterinario
  actualizarVeterinario(veterinarioId: number, veterinario: any): Observable<any> {
    return this.http.put<any>(`${this.ROOT_URL}/update/${veterinarioId}`, veterinario);
  }

  //Obtener tratamientos por el ID del veterinario
  getTratamientosByVeterinarioId(veterinarioId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.ROOT_URL}/tratamientos/${veterinarioId}`);
  }

  //Obtener cantidad de veterinarios activos
  getVeterinariosActivos(): Observable<any> {
    return this.http.get<any[]>(`${this.ROOT_URL}/veterinariosActivos`);
  }

  //Obtener cantidad de veterinarios inactivos
  getVeterinariosInactivos(): Observable<any> {
    return this.http.get<any[]>(`${this.ROOT_URL}/veterinariosInactivos`);
  }

  // Verificar inicio de sesión de un veterinario
  verificarInicioSesion(user:Veterinario): Observable<string> {
    return this.http.post(`${this.ROOT_URL}/login`, user,
      {
        responseType: "text"
      });
  }

  veterinarioHome():Observable<Veterinario>{
    return this.http.get<Veterinario>("http://localhost:8090/catcare/veterinarios/details");
  }

}




